test/open_api/map_test.exs

```diff
@@ -0,0 +1,403 @@
+defmodule OpenApiSpex.OpenApi.MapTest do
+  use ExUnit.Case
+  use Plug.Test
+
+  alias OpenApiSpex.OpenApi
+
+  describe "OpenApiSpex.OpenApi.Map.decode/1" do
+    test "OpenApi" do
+      # NOTE: This test could be split into many smaller tests, that is the goal!
+      # TODO: Move the spec to a setup below, where we could easily try a yaml version also
+      spec =
+        "./test/support/encoded_schema.json"
+        |> File.read!()
+        |> Jason.decode!(keys: :atoms)
```

Comment:
About this automatic conversion of string keys to atoms, this is typically viewed as an unsafe operation, as @josevalim detailed. [`String.to_atom/1` is being called somewhere in this `Jason.decode!/2` function](https://github.com/michalmuskala/jason/blob/master/lib/decoder.ex#L59). By only accepting maps with atom keys, `decode/1` is forcing the caller to perform an unsafe operation. Also, there will be situations where the caller cannot easily convert string keys to atom keys like this `Jason.decode!/2` does. Maybe the data is coming from an REST endpoint request, and the HTTP client automatically decodes the incoming JSON data. Or maybe the data is an XML parser.

Granted, the `OpenApi.Map` module does call `String.to_atom/1` in two places, but maybe we can find a solution to eliminate that.

I think this function would be significantly more valuable of it accepted maps with strings keys.

Reply:
This function is like `Kernel.struct/2`, except that will convert string keys to atoms: https://github.com/moxley/open_api_parse/blob/master/lib/open_api_parse/utils.ex

It can be a drop-in replacement for the `struct/2` calls.